### PR TITLE
Parse any `BinaryFloatingPoint` type

### PR DESCRIPTION
**Pull request by stephencelis**
We were originally piggybacking off the family of `strto*` functions for incrementally parsing floating point numbers off of strings. This pushed us to have dedicated parsers per type.

This changed in #5, but we never thought to squash all these parsers into one by abstracting over `BinaryFloatingPoint` instead. So this PR does just that.

### CODE REVIEW

File: Sources/Parsing/Parsers/Float.swift [hunk 1]
```diff
@@ -0,0 +1,172 @@
+extension BinaryFloatingPoint where Self: LosslessStringConvertible {
```
Comment (stephencelis) [inline Sources/Parsing/Parsers/Float.swift:1]: I moved this file to be called "Float.swift" now that it's generic over a "float"-named protocol. Do we want to rename "Double.md" documentation, too? Or should we be consistent and just prefer "Double" everywhere?

Comment (stephencelis) [inline Sources/Parsing/Parsers/Float.swift:1]: Worth noting that we need to abstract over `BinaryFloatingPoint` _and_ `LosslessStringConvertible` here. While `FixedWidthInteger` inherits from it automatically, `BinaryFloatingPoint` does not.

File: Sources/Parsing/Parsers/Float.swift [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    @inlinable
+    public init() {}
+
+    @inlinable
+    public func parse(_ input: inout Input) throws -> Output {
+      let original = input
+      let s = input.parseFloat()
+      guard let n = Output(String(decoding: s, as: UTF8.self))
+      else {
+        throw ParsingError.expectedInput("\(Output.self)".lowercased(), from: original, to: input)
```
Comment (stephencelis) [inline Sources/Parsing/Parsers/Float.swift:78]: Errors have changed slightly: `expected double` and `expected float` are unchanged, but if you use `Float80` you'll get `expected float80` now.

**Comment by stephencelis** [comment]
Also worth noting that in abstracting we now have a `Float16.parser()` for free...

@mbrandonw `Foundation.Decimal` does _not_ inherit from `BinaryFloatingPoint`, so we still don't have a `Decimal.parser()`...is that worth adding?

File: Sources/Parsing/Parsers/Float.swift (hunk 1 above)
Comment (stephencelis) [reply Sources/Parsing/Parsers/Float.swift:1]: - [ ] Remove `Double.md` and mention that `Double` works in `Float.md`
- [ ] Mention that other integer types work in `Int.md`

**Review by mbrandonw — APPROVED** [review]

**Pull request merged**